### PR TITLE
Create wildcard_string_matching.cpp

### PR DESCRIPTION
**Pull request by SumukhBhardwaj**
Added wildcard string matching problem, solved using dynamic programming

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1248"></a>

### CODE REVIEW

**Comment by SumukhBhardwaj** [comment]
@kvedala I have done all the changes please review it now.

File: dynamic_programming/wildcard_string_matching.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                return dpTable[pos1][pos2] = wildcardMatching(dpTable, s, p, pos1, pos2 + 1) || wildcardMatching(dpTable, s, p, pos1 + 1, pos2);
+	    }
+	    // not possible to match
+            else {
+                return dpTable[pos1][pos2] = 0;
+	    }
+        }
+}
+
+static void test() {
```
Comment (Panquesito7) [inline dynamic_programming/wildcard_string_matching.cpp:89]:
```suggestion
/**
 * @brief Test implementations
 * @returns void
 */
static void test() {
```

File: dynamic_programming/wildcard_string_matching.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#include <cassert>
+
+/**
+ * @brief This function implements the above algorithm
+ * @s matching string which is to be matched with pattern
+ * @p pattern string containing alphabets, '?' and '*'
+ * @pos1 the current position on string s
+ * @pos2 the current position on string p
+ * @dpTable is the memoization table, having the previously calculated result
+ * @returns 0/1 depending upon strings can be matched or not
```
Comment (Panquesito7) [inline dynamic_programming/wildcard_string_matching.cpp:36]:
```suggestion
 * @param s matching string which is to be matched with pattern
 * @param p pattern string containing alphabets, '?' and '*'
 * @param pos1 the current position on string s
 * @param pos2 the current position on string p
 * @param dpTable is the memorization table, having the previously calculated result
 * @returns 0/1 depending upon strings can be matched or not
```

File: dynamic_programming/wildcard_string_matching.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
+/*
```
Comment (Panquesito7) [inline dynamic_programming/wildcard_string_matching.cpp:1]: For how to document the first comment block, please see the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/SumukhBhardwaj/C-Plus-Plus/actions

**Pull request closed without merging**